src/BigDecimal.ts

```diff
@@ -0,0 +1,759 @@
+ * import { remainder, make } from 'effect/BigDecimal'as
+ * import { some, none } from 'effect/Option'
+ *
+ * assert.deepStrictEqual(remainder(make(2), make(2)), some(make(0)))
+ * assert.deepStrictEqual(remainder(make(3), make(2)), some(make(1)))
+ * assert.deepStrictEqual(remainder(make(-4), make(2)), some(make(0)))
+ *
+ * @since 2.0.0
+ * @category math
+ */
+export const remainder: {
+  (divisor: BigDecimal): (self: BigDecimal) => Option.Option<BigDecimal>
+  (self: BigDecimal, divisor: BigDecimal): Option.Option<BigDecimal>
+} = dual(2, (self: BigDecimal, divisor: BigDecimal): Option.Option<BigDecimal> => {
+  if (divisor.value === bigint0) {
+    return Option.none()
+  }
+
+  const max = Math.max(self.scale, divisor.scale)
+  return Option.some(scaled(scale(self, max).value % scale(divisor, max).value, max))
+})
+
+/**
+ * Returns the remainder left over when one operand is divided by a second operand.
+ *
+ * Throws a `RangeError` if the divisor is `0`.
+ *
+ * @param self - The dividend.
+ * @param divisor - The divisor.
+ *
+ * @example
+ * import { unsafeRemainder, make } from 'effect/BigDecimal'
+ *
+ * assert.deepStrictEqual(unsafeRemainder(make(2), make(2)), make(0))
+ * assert.deepStrictEqual(unsafeRemainder(make(3), make(2)), make(1))
+ * assert.deepStrictEqual(unsafeRemainder(make(-4), make(2)), make(0))
+ *
+ * @since 2.0.0
+ * @category math
+ */
+export const unsafeRemainder: {
+  (divisor: BigDecimal): (self: BigDecimal) => BigDecimal
+  (self: BigDecimal, divisor: BigDecimal): BigDecimal
+} = dual(2, (self: BigDecimal, divisor: BigDecimal): BigDecimal => {
+  if (divisor.value === bigint0) {
+    throw new RangeError("Division by zero")
+  }
+
+  const max = Math.max(self.scale, divisor.scale)
+  return scaled(scale(self, max).value % scale(divisor, max).value, max)
+})
+
+/**
+ * @category instances
+ * @since 2.0.0
+ */
+export const Equivalence: equivalence.Equivalence<BigDecimal> = equivalence.make((self, that) => {
+  if (self.scale > that.scale) {
+    return scale(that, self.scale).value === self.value
+  }
+
+  if (self.scale < that.scale) {
+    return scale(self, that.scale).value === that.value
+  }
+
+  return self.value === that.value
+})
+
+/**
+ * Checks if two `BigDecimal`s are equal.
+ *
+ * @since 2.0.0
+ * @category predicates
+ */
+export const equals: {
+  (that: BigDecimal): (self: BigDecimal) => boolean
+  (self: BigDecimal, that: BigDecimal): boolean
+} = dual(2, (self: BigDecimal, that: BigDecimal): boolean => Equivalence(self, that))
+
+/**
+ * Formats a given `BigDecimal` as a `string`.
+ *
+ * @param n - The `BigDecimal` to format.
+ *
+ * @example
+ * import { format, make } from 'effect/BigDecimal'
+ *
+ * assert.deepStrictEqual(format(make(-5n)), '-5')
+ * assert.deepStrictEqual(format(make(123.456)), '123.456')
+ * assert.deepStrictEqual(format(make(-0.00000123)), '-0.00000123')
+ *
+ * @since 2.0.0
+ * @category constructor
+ */
+export const format = (n: BigDecimal): string => {
```

Comment:
For interop with the js ecosystem, an `unsafeToNumber` could be useful.

Reply:
Simply as `(_) => Number(format(_))`?

Reply:
Yeah I guess that is the best way, otherwise unsafely casting the bigint then scaling it down.

Reply:
Can't scale down if it has a negative scale (cut off trailing zeros)